x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapperTests.java

```diff
@@ -174,42 +175,41 @@ public void testUpdatesToInferenceIdNotSupported() throws IOException {
     }
 
     public void testDynamicUpdate() throws IOException {
+        final String fieldName = "semantic";
+        final String inferenceId = "test_service";
+
         MapperService mapperService = createMapperService(mapping(b -> {}));
         mapperService.merge(
             "_doc",
             new CompressedXContent(
-                Strings.toString(PutMappingRequest.simpleMapping("semantic", "type=semantic_text,inference_id=test_service"))
+                Strings.toString(PutMappingRequest.simpleMapping(fieldName, "type=semantic_text,inference_id=" + inferenceId))
             ),
             MapperService.MergeReason.MAPPING_UPDATE
         );
-        String source = """
-            {
-              "semantic": {
-                "inference": {
-                  "inference_id": "test_service",
-                  "model_settings": {
-                    "task_type": "SPARSE_EMBEDDING"
-                  },
-                  "chunks": [
-                    {
-                      "embeddings": {
-                        "feature_0": 1
-                      },
-                      "text": "feature_0"
-                    }
-                  ]
-                }
-              }
-            }
-            """;
-        SourceToParse sourceToParse = new SourceToParse("test", new BytesArray(source), XContentType.JSON);
+
+        SemanticTextField semanticTextField = new SemanticTextField(
+            fieldName,
+            List.of(),
+            new SemanticTextField.InferenceResult(
+                inferenceId,
+                new SemanticTextField.ModelSettings(TaskType.SPARSE_EMBEDDING, null, null),
+                List.of()
+            ),
+            XContentType.JSON
+        );
+        XContentBuilder builder = JsonXContent.contentBuilder().startObject();
+        builder.field(semanticTextField.fieldName());
+        builder.value(semanticTextField);
+        builder.endObject();
+
+        SourceToParse sourceToParse = new SourceToParse("test", BytesReference.bytes(builder), XContentType.JSON);
```

Comment:
@jimczi I changed this test a bit to use an `XContentBuilder` to build source. This approach is more maintainable IMO because changes to the semantic text mapping API will either be automatically picked up or cause a build failure.

WDYT?